packages/client-sdk-nodejs/src/internal/storage-data-client.ts

```diff
@@ -111,9 +114,31 @@ export class StorageDataClient implements IStorageDataClient {
       new Header('agent', `nodejs:store:${version}`),
       new Header('runtime-version', `nodejs:${process.versions.node}`),
     ];
+
+    // Determine which retry strategy is specified in the configuration
+    // and which interceptor to use.
+    const retryStrategy = this.configuration.getRetryStrategy();
+    let timeoutInterceptor: Interceptor;
+    if (retryStrategy instanceof DefaultStorageRetryStrategy) {
+      const responseDataReceivedTimeoutMs =
+        retryStrategy.getResponseDataReceivedTimeoutMillis();
+      timeoutInterceptor = createStorageClientTimeoutInterceptor(
+        _loggerFactory,
+        this.requestTimeoutMs,
+        responseDataReceivedTimeoutMs
+      )[0];
+    } else {
+      timeoutInterceptor = ClientTimeoutInterceptor(this.requestTimeoutMs);
+    }
```

Comment:
oh whoops, saw your comment just after I'd pushed a fix for that. I made the internal `initializeInterceptors` function choose which interceptor to add based on the type of the retry strategy, but I can look into creating a single interceptor instead